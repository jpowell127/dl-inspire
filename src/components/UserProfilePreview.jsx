import React from "react";
import { Button, Card, Image } from "semantic-ui-react";

function UserProfilePreview({ user }) {
  const { first_name, last_name, username, url } = user;
  const image = user.images[50];
  return (
    <Card>
      <Card.Content>
        <Image
          floated="right"
          size="mini"
          src={image}
          alt={`Image of ${username}`}
        />
        <Card.Header>{first_name + " " + last_name}</Card.Header>
        <Card.Meta>{username}</Card.Meta>
        <Card.Description>
          <a href={url} target="_blank">
            {url}
          </a>
        </Card.Description>
      </Card.Content>
      <Card.Content extra>
        <div className="ui two buttons">
          <Button basic color="green">
            View
          </Button>
        </div>
      </Card.Content>
    </Card>
  );
}

export default UserProfilePreview;
