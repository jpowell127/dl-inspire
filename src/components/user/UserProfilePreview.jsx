import React from "react";
import PropTypes from "prop-types";
import { Button, Card, Image } from "semantic-ui-react";
import UserProfileModal from "./UserProfileModal";

export default function UserProfilePreview({ user }) {
  const { display_name, username, url, id } = user;
  const image = user.images[50];

  return (
    <Card color="pink">
      <Card.Content>
        <Image
          floated="right"
          size="mini"
          src={image}
          alt={`Image of ${username}`}
        />
        <Card.Header>{display_name}</Card.Header>
        <Card.Meta>{username}</Card.Meta>
      </Card.Content>
      <Card.Content extra>
        <UserProfileModal user={user} />
      </Card.Content>
    </Card>
  );
}

UserProfileModal.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    display_name: PropTypes.string.isRequired,
    username: PropTypes.string.isRequired,
    occupation: PropTypes.string,
    company: PropTypes.string,
    location: PropTypes.string,
    images: PropTypes.object.isRequired,
    stats: PropTypes.shape({
      followers: PropTypes.number.isRequired,
      following: PropTypes.number.isRequired,
      appreciations: PropTypes.number.isRequired,
      views: PropTypes.number.isRequired,
      comments: PropTypes.number.isRequired
    }).isRequired
  }).isRequired
};
