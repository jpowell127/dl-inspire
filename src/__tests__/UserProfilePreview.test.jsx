import React from "react";
import UserProfilePreview from "../components/user/UserProfilePreview";
import MockUser from "./mocks/mock.user";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";

test("<UserProfilePreview/> renders correctly", () => {
  const component = mount(<UserProfilePreview user={MockUser} />);
  expect(component.props().user).toBe(MockUser);
  expect(toJson(component)).toMatchSnapshot();
  component.unmount();
});
