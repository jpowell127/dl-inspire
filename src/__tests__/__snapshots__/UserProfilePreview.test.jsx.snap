// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfilePreview /> renders correctly 1`] = `
<UserProfilePreview
  user={
    Object {
      "city": "",
      "company": "",
      "country": "",
      "created_on": 0,
      "display_name": "",
      "fields": Array [],
      "first_name": "",
      "has_default_image": 0,
      "id": 12345,
      "images": Object {},
      "last_name": "",
      "location": "",
      "occupation": "",
      "state": "",
      "stats": Object {
        "appreciations": 0,
        "comments": 0,
        "followers": 0,
        "following": 0,
        "views": 0,
      },
      "url": "",
      "username": "",
      "website": "",
    }
  }
>
  <Card
    color="pink"
  >
    <div
      className="ui pink card"
      onClick={[Function]}
    >
      <CardContent>
        <div
          className="content"
        >
          <Image
            alt="Image of "
            as="img"
            floated="right"
            size="mini"
            ui={true}
          >
            <img
              alt="Image of "
              className="ui mini right floated image"
            />
          </Image>
          <CardHeader>
            <div
              className="header"
            />
          </CardHeader>
          <CardMeta>
            <div
              className="meta"
            />
          </CardMeta>
        </div>
      </CardContent>
      <CardContent
        extra={true}
      >
        <div
          className="extra content"
        >
          <UserProfileModal
            user={
              Object {
                "city": "",
                "company": "",
                "country": "",
                "created_on": 0,
                "display_name": "",
                "fields": Array [],
                "first_name": "",
                "has_default_image": 0,
                "id": 12345,
                "images": Object {},
                "last_name": "",
                "location": "",
                "occupation": "",
                "state": "",
                "stats": Object {
                  "appreciations": 0,
                  "comments": 0,
                  "followers": 0,
                  "following": 0,
                  "views": 0,
                },
                "url": "",
                "username": "",
                "website": "",
              }
            }
          >
            <Modal
              closeIcon={true}
              closeOnDimmerClick={true}
              closeOnDocumentClick={false}
              dimmer={true}
              eventPool="Modal"
              trigger={
                <Button
                  as="button"
                  color="pink"
                  fluid={true}
                  inverted={true}
                  role="button"
                >
                  View User
                </Button>
              }
            >
              <Portal
                className="ui page modals dimmer transition visible active"
                closeOnDocumentClick={false}
                closeOnEscape={true}
                closeOnRootNodeClick={true}
                eventPool="Modal"
                mountNode={<body />}
                onClose={[Function]}
                onMount={[Function]}
                onOpen={[Function]}
                onUnmount={[Function]}
                openOnTriggerClick={true}
                trigger={
                  <Button
                    as="button"
                    color="pink"
                    fluid={true}
                    inverted={true}
                    role="button"
                  >
                    View User
                  </Button>
                }
              >
                <Ref
                  innerRef={[Function]}
                >
                  <Button
                    as="button"
                    color="pink"
                    fluid={true}
                    inverted={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    role="button"
                  >
                    <button
                      className="ui pink fluid inverted button"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      role="button"
                    >
                      View User
                    </button>
                  </Button>
                </Ref>
              </Portal>
            </Modal>
          </UserProfileModal>
        </div>
      </CardContent>
    </div>
  </Card>
</UserProfilePreview>
`;
